lib/NormalModuleFactory.js

```diff
@@ -577,6 +578,15 @@ class NormalModuleFactory extends ModuleFactory {
 		resolveContext,
 		callback
 	) {
+		for (const item of array) {
```

Comment:
This check is at a pretty hot location. I would prefer to have it in `lib/rules/UseEffectRulePlugin.js` instead, to check it only only during parsing of the rules